Examples/Io/Root/src/RootMaterialTrackReader.cpp

```diff
@@ -89,11 +90,24 @@ ActsExamples::ProcessCode ActsExamples::RootMaterialTrackReader::read(
     // The collection to be written
     std::vector<Acts::RecordedMaterialTrack> mtrackCollection;
 
+    // Function to find the first entry that has the event number as executed
+    auto getEntry = [&]() -> unsigned int {
+      for (unsigned int j = 0; j < m_inputChain->GetEntries(); ++j) {
+        m_inputChain->GetEntry(j);
+        if (m_eventId == context.eventNumber) {
+          return j;
+        }
+      }
+      return m_cfg.batchSize * context.eventNumber;
+    };
+
+    auto eventFirstEntry = getEntry();
     for (size_t ib = 0; ib < m_cfg.batchSize; ++ib) {
```

Comment:
Instead of using batchSize that demand the users know the number of track per event wouldn't it be simpler to loop for as long as the event nb does change ?

Reply:
Good point. I removed the `batchSize` from the config. This info is now obtained by `GetEntries(...)`.

Reply:
I never though of using GetEntries that way is quite smart !